scripts/enforce-settings-gradle-match-directories

```diff
@@ -6,7 +6,7 @@ self_dir=$(cd "$(dirname "$0")" &>/dev/null; pwd -P)
 source "$self_dir"/utils.sh
 
 diff -u \
-  <(git ls-files */ | cut -d/ -f1 | uniq | grep -Exv '\..*|build|config|docs|gradle|.*\.hprof\..*|src|scripts') \
+  <(git ls-files --recurse-submodules */ | uniq | grep -E 'build.gradle'  | awk '$1=$1' FS="/" OFS=":" | awk '{gsub("build.gradle", "");print}' | sed 's/.$//' |  sed  's/gitCore:jGit/&\nqual/g') \
```

Comment:
`--recurse-submodules` recurses into submodules in the sense of [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules), which we don't keep/need in this project (in fact, I've only ever worked on two repos in one project that used submodules 😅). Not a bug per se, but redundant here

Reply:
Check

```shell
diff -u <(git ls-files --recurse-submodules */) <(git ls-files */)
```

to see there's no difference (see https://virtuslab.github.io/tips/#shell/output-comparing for `<(...)` syntax)

Reply:
Going further... the entire part 
```
git ls-files */ | uniq | grep -E 'build.gradle'
```

needed to list build.gradle files located **outside** the main directory can be simplified to:

```
git ls-files '**/build.gradle'
```

(again, process substitution for comparing the outputs FTW)

Reply:
Then another step:

```
... | awk '$1=$1' FS="/" OFS=":" | awk '{gsub("build.gradle", "");print}' | sed 's/.$//'
```

this can be replaced with less esoteric `tr` (not that `awk` is completely absent from this repo 😅 but for operations **within a single line** `sed`, `cut`, `tr` are usually better alternatives:

```
... | tr / : | sed 's/:build.gradle//'
```

Reply:
And finally, if it comes to missing build.gradle in `qual`... I think an empty build.gradle can be added there regardless of this script, just to make it clear that it's a regular Gradle module (even though it doesn't require any dependencies/special config)